app/controllers/iep_documents_controller.rb

```diff
@@ -1,37 +1,27 @@
 class IepDocumentsController < ApplicationController
 
-  before_action :find_iep_document, :authorize!
-
   def show
-    @file_name = @iep_document.file_name
-    @bucket_name = ENV['AWS_S3_IEP_BUCKET']
-
-    object = s3.get_object({
-      bucket: @bucket_name,
-      key: @file_name
-    })
-
-    send_data object.body.read, filename: @file_name, type: :pdf
+    iep_document = IepDocument.find_by_id!(params[:id])
+    authorized_or_raise! { iep_document.student }
```

Comment:
This also unrolls the control flow so it's explicit, rather than using `before_action` and split across several methods sharing instance variables.